include/knowhere/index/handle.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+#ifndef KNOWHERE_HANDLE_H
+#define KNOWHERE_HANDLE_H
+#include <atomic>
+#include <chrono>
+#include <memory>
+
+#include "knowhere/expected.h"
+
+namespace folly {
+template <typename T>
+class Future;
+
+}
+
+namespace knowhere {
+class Handle {
+ public:
+#ifdef KNOWHERE_WITH_CARDINAL
```

Comment:
why is it needed to add pragmas for `KNOWHERE_WITH_CARDINAL` for this particular case at all? I think it can be avoided by just ignoring timeouts in non-cardinal based implementations

Reply:
The asynchronous interface is universally supported, but only Cardinal supports the timeout and proactive stop mechanisms.